#!/usr/bin/env python
PACKAGE = "perception"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("leafsize",    double_t,    0, "leafsize parameter", 0.0095,  0.001, 0.1)
gen.add("filter_mean", double_t, 0, "filter_mean parameter",    10.0, 0,   80)
gen.add("filter_thresold",    double_t,    0, "filter_thresold parameter",  0.2, 0,   10)
gen.add("segmentation_thresold",   double_t,   0, "segmentation_thresold parameter",  0.01,0,5)
gen.add("segmentation_maxiteration",   int_t,   0, "segmentation_maxiteration parameter",  50,0,1000)
gen.add("ClusterTolerance",   double_t,   0, "ClusterTolerance parameter",  0.03,0,0.1)
gen.add("ClusterMinSize",   int_t,   0, "ClusterMinSize parameter",  20,0,500)
gen.add("ClusterMaxSize",   int_t,   0, "ClusterMinSize parameter",  45,0,500)
gen.add("passFilterMin_x",    double_t,    0, "passFilterMin_x parameter", -0.05,  -1, 1)
gen.add("passFilterMax_x",    double_t,    0, "passFilterMax_x parameter", 0.4, -1, 1)
gen.add("passFilterMin_y",    double_t,    0, "passFilterMin_y parameter", -0.2,  -1, 1)
gen.add("passFilterMax_y",    double_t,    0, "passFilterMax_y parameter", 0.2, -1, 1)
gen.add("passFilterMin_z",    double_t,    0, "passFilterMin_z parameter", 0.84,  -1, 1)
gen.add("passFilterMax_z",    double_t,    0, "passFilterMax_z parameter", 0.87, -1, 1)
gen.add("convexHullHeightMin",  double_t,   0,  "convexHullHeightMin parameter", -0.02, -1, 1)
gen.add("convexHullHeightMax",  double_t,   0,  "convexHullHeightMax paremeter", 0.005, -1, 1)

exit(gen.generate(PACKAGE, "perception", "reconfig")) 
